### PR TITLE
Port code to Puppet 7 and 8

### PR DESCRIPTION
**Pull request by Valantin**
#### This Pull Request (PR) fixes the following issues

include #6

### CODE REVIEW

**Comment by traylenator** [comment]
We really need #3 which needs a vagrant-beaker release before we can add some sensible OSes.

File: manifests/files.pp [hunk 1]
```diff
@@ -33,15 +33,15 @@
 #
 define swap_file::files (
   Enum['absent','present'] $ensure = 'present',
-  Stdlib::Unixpath $swapfile       = '/mnt/swap.1',
+  Stdlib::Absolutepath $swapfile   = '/mnt/swap.1',
   $swapfilesize                    = $facts['memory']['system']['total'],
   Boolean $add_mount               = true,
-  $options                         = 'defaults',
+  String $options                  = 'defaults',
```
Comment (traylenator) [inline manifests/files.pp:39]: This is good but we should re-enable the tests which are disabled in puppet-lint.rc or whatever it is... Actually am a bit confused why the modulesync did not fix the file.

**Comment by sahaqaa** [comment]
Hello, please advice, is there any chance to get this PR merged? Thank you.

Looking forward to seeing first release of this module to be released to Forge.

**Comment by traylenator** [comment]
Not needed anymore.

**Pull request closed without merging**